cascade/utils.py

```diff
@@ -0,0 +1,48 @@
+from io import StringIO
+import ase
+from ase import Atoms
+
+# Taken from ExaMol
+# Taken from jitterbug
+def write_to_string(atoms: Atoms, fmt: str, **kwargs) -> str:
+    """Write an ASE atoms object to string
+
+    Args:
+        atoms: Structure to write
+        fmt: Target format
+        kwargs: Passed to the write function
+    Returns:
+        Structure written in target format
+    """
+
+    out = StringIO()
+    atoms.write(out, fmt, **kwargs)
+    return out.getvalue()
+
+def read_from_string(atoms_msg: str, fmt: str) -> Atoms:
+    """Read an ASE atoms object from a string
+
+    Args:
+        atoms_msg: String format of the object to read
+        fmt: Format (cannot be autodetected)
+    Returns:
+        Parsed atoms object
+    """
+
+    out = StringIO(str(atoms_msg))  # str() ensures that Proxies are resolved
+    return ase.io.read(out, format=fmt)
+
+def canonicalize(atoms: Atoms, fmt: str = 'extxyz') -> Atoms:
```

Comment:
Let's keep this for now, but I'm curious whether we should just directly make the `SinglePointCalculator` rather than this indirect route. 

Also, I'd remove `fmt` as an argument if the code works as desired with extxyz and not others

Reply:
(This = direct conversion)